pkg/operator/starter_test.go

```diff
@@ -0,0 +1,232 @@
+				{
+					Name: "rc",
+					URL:  "https://rc.ibmcloud-test.com",
+				},
+				{
+					Name: "pi",
+					URL:  "https://pi.ibmcloud-test.com",
+				},
+			},
+			inDeployment: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{{
+								Name: "csi-driver",
+								Env: []corev1.EnvVar{
+									{
+										Name:  "IBMCLOUD_APIKEY",
+										Value: "testKey",
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+			expected: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{{
+								Name: "csi-driver",
+								Env: []corev1.EnvVar{
+									{
+										Name:  "IBMCLOUD_APIKEY",
+										Value: "testKey",
+									},
+									{
+										Name:  "IBMCLOUD_IAM_API_ENDPOINT",
+										Value: "https://iam.ibmcloud-test.com",
+									},
+									{
+										Name:  "IBMCLOUD_RESOURCE_CONTROLLER_API_ENDPOINT",
+										Value: "https://rc.ibmcloud-test.com",
+									},
+									{
+										Name:  "IBMCLOUD_POWER_API_ENDPOINT",
+										Value: "https://pi.ibmcloud-test.com",
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "when custom endpoints are specified along with a required unmapped field",
+			customEndPoints: []v1.PowerVSServiceEndpoint{
+				{
+					Name: "iam",
+					URL:  "https://iam.ibmcloud-test.com",
+				},
+				{
+					Name: "rc",
+					URL:  "https://rc.ibmcloud-test.com",
+				},
+				{
+					Name: "pi",
+					URL:  "https://pi.ibmcloud-test.com",
+				},
+				{
+					Name: "IBMCLOUD_COS_ENDPOINT",
+					URL:  "https://s3.us-south.cloud-object-storage.appdomain.cloud",
+				},
+			},
+			inDeployment: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{{
+								Name: "csi-driver",
+								Env: []corev1.EnvVar{
+									{
+										Name:  "IBMCLOUD_APIKEY",
+										Value: "testKey",
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+			expected: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{{
+								Name: "csi-driver",
+								Env: []corev1.EnvVar{
+									{
+										Name:  "IBMCLOUD_APIKEY",
+										Value: "testKey",
+									},
+									{
+										Name:  "IBMCLOUD_IAM_API_ENDPOINT",
+										Value: "https://iam.ibmcloud-test.com",
+									},
+									{
+										Name:  "IBMCLOUD_RESOURCE_CONTROLLER_API_ENDPOINT",
+										Value: "https://rc.ibmcloud-test.com",
+									},
+									{
+										Name:  "IBMCLOUD_POWER_API_ENDPOINT",
+										Value: "https://pi.ibmcloud-test.com",
+									},
+									{
+										Name:  "IBMCLOUD_COS_ENDPOINT",
```

Comment:
Does this endpoint matter for CSI driver?

Reply:
I don't think apart from `IBMCLOUD_RESOURCE_CONTROLLER_API_ENDPOINT`, `IBMCLOUD_POWER_API_ENDPOINT` and `IBMCLOUD_IAM_API_ENDPOINT`, anything else would be optional. The driver, at the moment depends only on the above three. 

Should we consider limiting the scope?

Reply:
Will reduce the scopw as discussed. Thanks!